### PR TITLE
Reprise globale du package

### PR DESCRIPTION
**Pull request by maniackcrudelis**
Très gros refactoring du code.

- Passe tout les tests sur package_check
- Upgrade testé depuis la version précédente.

Dans les grandes lignes:

- is_public et multisite passent en boolean dans le manifest
- Usage de app.src pour la récupération des sources
- User dédié au lieu de www-data
- set -eu ou ynh_abort_if_errors
- .fonctions dans backup et restore
- Usage des helpers YunoHost
- Clarification des scripts
- path_url au lieu de path
- systemctl au lieu de system
- Usage de ynh_install_app_dependencies
- Backup before upgrade (Testé avec succès)

### CODE REVIEW

**Comment by Psycojoker** [comment]
This is like a super giganormous hardcore review you're asking right now (+951 −424, nearly a rewrite, I don't know for the others ones, but I personally don't have the time and especially the concentration to do this kind of review :/

File: scripts/install [hunk 1]
```diff
@@ -1,156 +1,196 @@
...
-sleep 1
-curl -kL -H "Host: $domain" --data "&weblog_title=YunoBlog&user_name=$admin_wordpress&admin_password=$db_pwd&admin_password2=$db_pwd&admin_email=$admin_wordpress@$domain&language=$language&Submit=Install+WordPress" https://localhost$path/wp-admin/install.php?step=2 > /dev/null 2>&1
+YNH_CURL "&weblog_title=YunoBlog&user_name=$admin_wordpress&admin_password=$db_pwd&admin_password2=$db_pwd&admin_email=$admin_wordpress@$domain&language=$language&Submit=Install+WordPress" "/wp-admin/install.php?step=2"
 
-echo -n "Please wait during Wordpress installation" >&2
+WARNING echo -n "Please wait during Wordpress installation"
 for i in `seq 1 300`
 do	# La boucle attend la fin de l'installation de wordpress Ou 5 minutes.
-	if mysql --debug-check -u $db_user -p$db_pwd $db_user -e "show tables" | grep -q "wp_options"; then
+	if ynh_mysql_connect_as $db_name $db_pwd $db_name -e "show tables" | grep -q "wp_options"; then
```
Comment (JimboJoe) [inline scripts/install]: This command hangs for me. Shouldn't the syntax be `ynh_mysql_connect_as... <<< "show tables;"` ?

Comment (maniackcrudelis) [reply scripts/install]: Yes you're right, that works on my CI, but not on a VM.
With this correction, that seems really better.

**Review by JimboJoe — APPROVED** [review]

**Comment by maniackcrudelis** [comment]
Hi guys

We have a [issue](https://forum.yunohost.org/t/install-of-wp-in-failed/2855) with wordpress, so can we take a look at this PR ?
I know this is a big PR, but it's a global refactoring.

**Comment by maniackcrudelis** [comment]
A little bit of braveness and love for an old PR ?
Today is its 2 month birthday ! Poor little PR...

**Comment by Zykino** [comment]
I was about to install Wordpress on my Yunohost instance and saw this PR.
Now I'm worried about the upgrade from now and after this PR is merged.

Will I have any problem as a user after the merge is done?
Will this "blocking" PR block Wordpress from being updated to latest version (security patch and all)?

**Comment by maniackcrudelis** [comment]
Don't worry, you can install the current version and do an upgrade from the admin panel of wordpress. In fact, to prevent any failure, wordpress manage its updates by himself.

It's more a refactoring of the scripts for YunoHost. But it's such a big refactoring...

However, we already check that the current wordpress can be upgrade from this PR.

So, let's go, don't worry.

**Pull request merged**